gremlin-server/src/test/java/org/apache/tinkerpop/gremlin/server/GremlinServerIntegrateTest.java

```diff
@@ -411,6 +412,34 @@ public void shouldTimeOutRemoteTraversal() throws Exception {
         }
     }
 
+    @Test
+    public void shouldTimeOutRemoteTraversalWithPerRequestOption() throws Exception {
```

Comment:
Would it be possible to add a unit test that verifies that when PER_REQUEST options are set within a GraphTraversal they effectively end up in the RequestMessage's RequestOptions?

Or instead just that they end up being correctly parsed in DriverRemoteConnection.submitAsync() somehow

Reply:
I added a unit test by extracting the `RequestOptions` to its own method on `DriverRemoteConnection`.  Method has `protected` scope which is a little weird, but it's the easiest way to test it.